x-pack/plugins/observability_solution/infra/public/components/asset_details/tabs/overview/overview.tsx

```diff
@@ -44,18 +65,45 @@ export const Overview = () => {
     />
   );
 
+  const shouldShowCallout = () => {
+    if (dataStreamsStatus !== 'success') {
+      return false;
+    }
+
+    const { type } = asset;
+    const baseCondition =
+      !isMetricsSignal(dataStreams) && !dismissedAddMetricsCallout && renderMode.mode === 'page';
+
+    const isRelevantContainer =
+      type === 'container' && (isDockerContainer || isKubernetesContainer);
+
+    return baseCondition && (type === 'host' || isRelevantContainer);
```

Comment:
Why can't we just check if there's metrics signal? Why do we need all these checks on `container` and `host`?

Reply:
You can find additional context in [this comment](https://github.com/elastic/kibana/issues/193703#issuecomment-2397260453) and the following one.